worker/src/checks/hardware/gpu.rs

```diff
@@ -72,30 +64,48 @@ fn get_gpu_status() -> GpuDevice {
     // Get device count
     let device_count = match nvml.device_count() {
         Ok(count) => count as usize,
-        Err(e) => return GpuDevice::NotAvailable(format!("Failed to get device count: {}", e)),
+        Err(e) => {
+            Console::error(&format!("Failed to get device count: {}", e));
+            return vec![];
+        }
     };
 
     if device_count == 0 {
-        return GpuDevice::NotAvailable("No GPU devices detected".to_string());
+        Console::error("No GPU devices detected");
+        return vec![];
     }
 
-    // Get first device info
-    // TODO: Get all devices
-    match nvml.device_by_index(0) {
-        Ok(device) => {
-            let name = device.name().unwrap_or_else(|_| "Unknown".to_string());
-            let memory = device.memory_info().map(|m| m.total).unwrap_or(0);
-            let driver_version = nvml
-                .sys_driver_version()
-                .unwrap_or_else(|_| "Unknown".to_string());
+    let mut device_map: std::collections::HashMap<String, GpuDevice> =
+        std::collections::HashMap::new();
+
+    for i in 0..device_count {
+        match nvml.device_by_index(i as u32) {
+            Ok(device) => {
+                let name = device.name().unwrap_or_else(|_| "Unknown".to_string());
+                let memory = device.memory_info().map(|m| m.total).unwrap_or(0);
+                let driver_version = nvml
+                    .sys_driver_version()
+                    .unwrap_or_else(|_| "Unknown".to_string());
 
-            GpuDevice::Available {
-                name,
-                memory,
-                driver_version,
-                device_count,
+                if let Some(existing_device) = device_map.get_mut(&name) {
+                    existing_device.count += 1;
+                } else {
+                    device_map.insert(
+                        name.clone(),
+                        GpuDevice {
+                            name,
+                            memory,
+                            driver_version,
+                            count: 1,
+                        },
+                    );
+                }
+            }
```

Comment:
Would be useful to change this data structure to contain the cuda indexes of these GPUs, so that when you create a docker container instead of passing `--gpus all` you could have the option to do `docker run --gpus '"device=3,4,5"'`, and as such removes the need to worry about handling random indexes for mixed card types from inside the worker containers.